vignettes/summary_selecting.Rmd

```diff
@@ -0,0 +1,229 @@
+---
+title: "A quick summary of selecting possibilities in `datawizard`"
+output: rmarkdown::html_vignette
+vignette: >
+  %\VignetteIndexEntry{A quick summary of selecting possibilities in `datawizard`}
+  %\VignetteEngine{knitr::rmarkdown}
+  %\VignetteEncoding{UTF-8}
+---
+
+```{r, include = FALSE}
+knitr::opts_chunk$set(
+  collapse = TRUE,
+  comment = "#>"
+)
+```
+
+```{r setup, echo=FALSE, message=FALSE}
+library(datawizard)
+library(poorman)
+library(htmltools)
+
+set.seed(123)
+iris <- iris[sample(nrow(iris), 10), ]
+row.names(iris) <- NULL
+
+row <- function(...) {
+  div(
+    class = "custom_note",
+    ...
+  )
+}
+```
+
+```{css, echo=FALSE}
+.custom_note {
+  border-left: solid 5px hsl(220, 100%, 30%);
+  background-color: hsl(220, 100%, 95%);
+  padding: 5px;
+  margin-bottom: 10px
+}
+```
+
+# Selecting variables
+
+## Quoted names
+
+This is the most simple way to select one or several variables. Just use a character
+vector containing variables names, like in base R.
+
+```{r}
+get_columns(iris, c("Sepal.Length", "Petal.Width"))
+```
+
+## Unquoted names
+
+It is also possible to use unquoted names. This is useful if we use the `tidyverse`
+and want to be consistent about the way variable names are passed.
+
+```{r}
+iris %>%
+  group_by(Species) %>%
+  standardise(Petal.Length) %>%
+  ungroup()
+```
+
+
+## Formulas
+
+You can also use formulas to select variables:
+
+```{r}
+get_columns(iris, ~ Sepal.Length + Petal.Width)
+```
+
+This was mostly useful to use selection more easily in custom functions before
+`datawizard` 0.6.0, and is kept available for backward compatibility. 
+
+
+
+## Positions
+
+In addition to variable names, `select` can also take a numeric vector that indicates
+the position of the variables to select in the dataframe.
+
+```{r}
+get_columns(iris, c(1, 2, 5))
+```
+
+
+## Functions
+
+We can pass a function to `select`. This function will be applied to all columns
+and has to return `TRUE` or `FALSE`. For example, if we want to keep only numeric
+columns, we can use `is.numeric`.
+
+```{r}
+get_columns(iris, is.numeric)
```

Comment:
Maybe we can also add an example on the negative case. That is, what the user needs to do if they want to select everything but the numeric type columns (`!is.numeric`).

Reply:
I think this is more suited in the section "Excluding variables". And actually, `!is.numeric` doesn't work (did it work before? it was not in the tests) but `-is.numeric` works

Reply:
Not sure if it worked before, but the easiest way here is to use a function operator:

``` r
library(datawizard)

get_columns(iris, Negate(is.numeric))
#>        Species
#> 1       setosa
#> 2       setosa
#> 3       setosa
#> 4       setosa
#> 5       setosa
#> 6       setosa
#> 7       setosa
#> 8       setosa
#> 9       setosa
#> 10      setosa
#> 11      setosa
#> 12      setosa
#> 13      setosa
#> 14      setosa
#> 15      setosa
#> 16      setosa
#> 17      setosa
#> 18      setosa
#> 19      setosa
#> 20      setosa
#> 21      setosa
#> 22      setosa
#> 23      setosa
#> 24      setosa
#> 25      setosa
#> 26      setosa
#> 27      setosa
#> 28      setosa
#> 29      setosa
#> 30      setosa
#> 31      setosa
#> 32      setosa
#> 33      setosa
#> 34      setosa
#> 35      setosa
#> 36      setosa
#> 37      setosa
#> 38      setosa
#> 39      setosa
#> 40      setosa
#> 41      setosa
#> 42      setosa
#> 43      setosa
#> 44      setosa
#> 45      setosa
#> 46      setosa
#> 47      setosa
#> 48      setosa
#> 49      setosa
#> 50      setosa
#> 51  versicolor
#> 52  versicolor
#> 53  versicolor
#> 54  versicolor
#> 55  versicolor
#> 56  versicolor
#> 57  versicolor
#> 58  versicolor
#> 59  versicolor
#> 60  versicolor
#> 61  versicolor
#> 62  versicolor
#> 63  versicolor
#> 64  versicolor
#> 65  versicolor
#> 66  versicolor
#> 67  versicolor
#> 68  versicolor
#> 69  versicolor
#> 70  versicolor
#> 71  versicolor
#> 72  versicolor
#> 73  versicolor
#> 74  versicolor
#> 75  versicolor
#> 76  versicolor
#> 77  versicolor
#> 78  versicolor
#> 79  versicolor
#> 80  versicolor
#> 81  versicolor
#> 82  versicolor
#> 83  versicolor
#> 84  versicolor
#> 85  versicolor
#> 86  versicolor
#> 87  versicolor
#> 88  versicolor
#> 89  versicolor
#> 90  versicolor
#> 91  versicolor
#> 92  versicolor
#> 93  versicolor
#> 94  versicolor
#> 95  versicolor
#> 96  versicolor
#> 97  versicolor
#> 98  versicolor
#> 99  versicolor
#> 100 versicolor
#> 101  virginica
#> 102  virginica
#> 103  virginica
#> 104  virginica
#> 105  virginica
#> 106  virginica
#> 107  virginica
#> 108  virginica
#> 109  virginica
#> 110  virginica
#> 111  virginica
#> 112  virginica
#> 113  virginica
#> 114  virginica
#> 115  virginica
#> 116  virginica
#> 117  virginica
#> 118  virginica
#> 119  virginica
#> 120  virginica
#> 121  virginica
#> 122  virginica
#> 123  virginica
#> 124  virginica
#> 125  virginica
#> 126  virginica
#> 127  virginica
#> 128  virginica
#> 129  virginica
#> 130  virginica
#> 131  virginica
#> 132  virginica
#> 133  virginica
#> 134  virginica
#> 135  virginica
#> 136  virginica
#> 137  virginica
#> 138  virginica
#> 139  virginica
#> 140  virginica
#> 141  virginica
#> 142  virginica
#> 143  virginica
#> 144  virginica
#> 145  virginica
#> 146  virginica
#> 147  virginica
#> 148  virginica
#> 149  virginica
#> 150  virginica
```

<sup>Created on 2022-09-15 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

Reply:
I don't think it's the easiest way. To me, we should stick to the "put a minus sign in front of any selection and the selection is reversed" approach

Reply:
Okay. TBH, it's not that crucial ATM, since the users do have the `exclude` argument to do this:

``` r
library(datawizard)

df <- tibble::as_tibble(iris)
data_select(df, exclude = is.numeric)
#> # A tibble: 150 × 1
#>    Species
#>  * <fct>  
#>  1 setosa 
#>  2 setosa 
#>  3 setosa 
#>  4 setosa 
#>  5 setosa 
#>  6 setosa 
#>  7 setosa 
#>  8 setosa 
#>  9 setosa 
#> 10 setosa 
#> # … with 140 more rows
```

<sup>Created on 2022-09-15 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>